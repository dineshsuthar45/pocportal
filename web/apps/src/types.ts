import { InputProps, TypographyProps } from "@mui/material";
import { CSSProperties, ChangeEvent } from "react";

export interface IThemeSchema {
  palette: any;
  typography: any;
  components?: any;
}

export interface IGlobalTableProps {
  row: any;
  list: any;
}

export interface IGlobalButtonProps {
  label: any;
  sublabel?: string;
  disabled?: boolean;
  onClick?: () => void;
  style?: CSSProperties;
  type?: string;
}

export interface IGlobalImgProps {
  img: string;
  alt: string;
  style?: CSSProperties;
}

export interface IGlobaltextFieldProps {
  id?: string;
  value: string;
  label: string;
  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
  name: string;
  errmsg?: string;
  type?: string;
  disabled?: boolean;
  style?: CSSProperties;
  placeholder?: string;
  showPassword?: boolean;
  InputProps?: InputProps;
}

export interface IGlobalTypoProps {
  id?: number;
  variant: TypographyProps["variant"];
  label: string;
  sublabel?: string;
  style?: CSSProperties;
}
