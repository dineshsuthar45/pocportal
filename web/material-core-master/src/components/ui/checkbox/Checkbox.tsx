import MuiCheckbox from "@mui/material/Checkbox";
import { ChangeEvent, useEffect, useState } from "react";
import { v4 as uuidv4 } from "uuid";
import { MFormControlLabel } from "../internal/formControlLabel";
import { CheckboxProps } from "./Checkbox.types";

/**
 * Further properties can be found in MUI website at https://mui.com/material-ui/api/checkbox/
 */
export const MCheckbox = ({
  label,
  labelPlacement = "end",
  tooltipTitle = "",
  checked = false,
  className = "",
  color = "secondary",
  disabled = false,
  disableRipple = false,
  id = uuidv4(),
  indeterminate = false,
  required = false,
  size = "small",
  ...props
}: CheckboxProps) => {
  const [internalTooltipTitle, setInternalTooltipTitle] = useState(
    tooltipTitle || label || ""
  );
  const [showAlwaysTooltip, setShowAlwaysTooltip] = useState(false);

  useEffect(() => {
    setInternalTooltipTitle(tooltipTitle || label || "");
  }, [tooltipTitle, label]);

  useEffect(() => {
    setShowAlwaysTooltip(!!tooltipTitle && tooltipTitle !== label);
  }, [label, tooltipTitle]);

  const handleClick = (event: unknown) => {
    if (!disabled && props.onChange) {
      props.onChange(event as ChangeEvent<HTMLInputElement>, !checked);
    }
  };

  return (
    <div
      onClick={handleClick}
      className={Array.isArray(className) ? className.join(" ") : className}
    >
      <MFormControlLabel
        label={label}
        labelPlacement={labelPlacement}
        tooltipProps={{
          title: internalTooltipTitle,
          showAlways: showAlwaysTooltip,
        }}
        disabled={disabled}
      >
        <MuiCheckbox
          id={id}
          checked={checked}
          color={color}
          disabled={disabled}
          disableRipple={disableRipple}
          indeterminate={indeterminate}
          required={required}
          size={size}
          {...props}
          onChange={() => {}}
        />
      </MFormControlLabel>
    </div>
  );
};

MCheckbox.defaultProps = {
  labelPlacement: "end",
  tooltipTitle: "",
  checked: false,
  className: "",
  color: "secondary",
  disabled: false,
  disableRipple: false,
  id: uuidv4(),
};
