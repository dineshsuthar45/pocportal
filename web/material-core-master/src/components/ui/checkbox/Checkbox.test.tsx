import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import React from "react";
import { MCheckbox } from "./Checkbox";
import { CheckboxProps } from "./Checkbox.types";

describe("Checkbox component", () => {
  const renderComponent = () => render(<MCheckbox {...props} />);
  let props: CheckboxProps;

  beforeEach(() => {
    props = {
      label: "I agree with T&Cs",
      id: "chk1",
      onChange: () => {},
    };
  });

  it("should render correctly", () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  it("disabled checkbox cannot be selected", () => {
    props.disabled = true;
    renderComponent();

    const checkbox = screen.getByRole<HTMLInputElement>("checkbox");
    expect(checkbox).toBeDisabled();
    expect(() => userEvent.click(checkbox)).toThrow();
  });
});
