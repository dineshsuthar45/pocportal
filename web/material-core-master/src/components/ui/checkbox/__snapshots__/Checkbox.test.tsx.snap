// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component should render correctly 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="m-form-control-label end"
    >
      <div
        aria-label=""
        class="m-tooltip-root tooltip-label tooltip-container"
        data-mui-internal-clone-element="true"
      >
        <label>
          I agree with T&Cs
        </label>
      </div>
      <div
        class="label-children"
      >
        <span
          class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorSecondary MuiCheckbox-sizeSmall PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorSecondary MuiCheckbox-sizeSmall MuiCheckbox-root MuiCheckbox-colorSecondary MuiCheckbox-sizeSmall css-1sg8fd1-MuiButtonBase-root-MuiCheckbox-root"
        >
          <input
            class="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate="false"
            id="chk1"
            type="checkbox"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;
